common-utils/src/main/groovy/org/orbisgis/common_utils/ConnectionUtils.groovy

```diff
@@ -0,0 +1,65 @@
+/*
+ * Bundle common-utils is part of the OrbisGIS platform
+ *
+ * OrbisGIS is a java GIS application dedicated to research in GIScience.
+ * OrbisGIS is developed by the GIS group of the DECIDE team of the
+ * Lab-STICC CNRS laboratory, see <http://www.lab-sticc.fr/>.
+ *
+ * The GIS group of the DECIDE team is located at :
+ *
+ * Laboratoire Lab-STICC – CNRS UMR 6285
+ * Equipe DECIDE
+ * UNIVERSITÉ DE BRETAGNE-SUD
+ * Institut Universitaire de Technologie de Vannes
+ * 8, Rue Montaigne - BP 561 56017 Vannes Cedex
+ *
+ * OSM is distributed under LGPL 3 license.
+ *
+ * Copyright (C) 2020 CNRS (Lab-STICC UMR CNRS 6285)
+ *
+ *
+ * OSM is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * OSM is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with
+ * OSM. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, please consult: <http://www.orbisgis.org/>
+ * or contact directly:
+ * info_at_ orbisgis.org
+ */
+package org.orbisgis.common_utils
+
+import groovy.sql.GroovyRowResult
+import groovy.sql.Sql
+
+import java.sql.Connection
+
+/**
+ * Utility script used as extension module adding methods to Connection class.
+ *
+ * @author Erwan Bocher (CNRS)
+ * @author Sylvain PALOMINOS (UBS chaire GEOTERA 2020)
+ */
+
+static boolean execute(Connection connection, String sql) {
+    return new Sql(connection).execute(sql)
+}
+
+static boolean execute(Connection connection, GString sql) {
+    return new Sql(connection).execute(sql)
+}
+
+static List<GroovyRowResult> rows(Connection connection, String sql) {
+    return new Sql(connection).rows((String)sql)
```

Comment:
Does make sense to load each time a sql object ?

Reply:
Last time we have discussed about it because previously the Sql object was cached but it seemed to be a bad idea if the datasource is managing a pool connection.

Reply:
Yes but in OrbisData with extend SQL to to limit object reconstruction.
Is it possible to evaluate the overhead if the user makes several rows calls ?

Reply:
In orbisdata, the Sql object is already cached so the issue is the same for both case. in my own opinion, the best thing to do is to cache the Sql object.

Reply:
I will be interested to have some explanations about the caching system you have in mind. ;-)

Reply:
Perhaps we can replace the row method by another one ?